README.txt
# $Id: README.txt,v 1.18.2.2.2.12.2.1 2009/12/07 17:32:59 designerbrent Exp $


--- README  -------------------------------------------------------------

SimplyCivi, Version 2.x

Brent Hardinge, aka, designerbrent
 designerbrent@gmail.com
 http://brenthardinge.net/
 
 
Contributors:

Originally by Ted Serbinski, aka, m3avrck
  hello@tedserbinski.com
  http://tedserbinski.com

Richard Burford, aka, psynaptic
 rich@freestylesystems.co.uk
 http://www.freestylesystems.co.uk

Requirements: Drupal 6.x, SimplyCivi Framework downloaded from SimplyCivicss.org


--- INSTALLATION --------------------------------------------------------

1. Place the SimplyCivi folder in your themes directory.

2. Download SimplyCivi http://www.SimplyCivicss.org/
   a. Extract folder, creating something like "joshuaclayton-SimplyCivi-css-28c8aa9ae2686442e00a5c7f46dfe2de76b3bd83"
   b. Rename to "framework"
   c. Ensure your path looks like themes/SimplyCivi/SimplyCivi/framework/SimplyCivi/screen.css

3. Enable theme under Administer > Site building > Themes


--- FEATURES --------------------------------------------------------

- uses SimplyCivi 0.8 (or greater) CSS framework: http://www.SimplyCivicss.org/
- normalizes Drupal's CSS to be consistent
- properly aggregates all SimplyCivi CSS files into a single file when this setting is enabled
- put scripts at bottom of page for nice performance gains, read more: http://developer.yahoo.com/performance/rules.html#js_bottom
- flexible layout, from 1 to 3 columns, based on where you configure your blocks to show (left, center, right)
- SEO optimization without the need for heavy modules and additional queries per page
  - automatically adds META description to many pages, read more: http://googlewebmastercentral.blogspot.com/2007/09/improve-snippets-with-meta-description.html
  - automatically adds META keywords if taxonomy exists on that node, read more: http://searchengineland.com/070905-194221.php
  - avoid duplicate titles in search indexes for pager pages: http://www.seo-expert-blog.com/blog/avoiding-duplicate-title-tags-on-pager-pages-in-drupal
- better forum icons, http://drupal.org/node/102743#comment-664157
- improve forum display and performance, http://www.sysarchitects.com/node/70
- prevents duplicate form submissions with jQuery, read more: http://tedserbinski.com/2007/01/11/how_to_prevent_duplicate_posts
  loading animation care of: http://www.ajaxload.info/
- shows the # of comments below a node since Drupal doesn't do this by default (usability)
- add permalinks to each comment (usability)
- highlight any comments by the author of the node
- adds a "you need to login / register" box below all comments if you can't add a comment (usability)
- supports conditional comment subjects, if the setting is off it won't show a chopped off title of the comment
- uses CSSEdit http://macrabbit.com/cssedit/ comments for grouping of styles
- lots of comments and theming tricks in template.php to learn from :)

Drupal 6 version:
- Hover over blocks to reveal admin links to edit and configure the blocks as well as edit the menu blocks.
- Support for sub-themes.

--- TIPS --------------------------------------------------------

- put your custom styles in css/style.css
- put any IE hacks into css/ie.css (conditionally loaded as needed)
- admin/build/themes/settings
  - enable site slogan (add one, good for SEO)
  - enable mission statement (used as META description for homepage in search engines)
  - enable user pictures
- Configure the comments for each node type 
  - admin/content/node-type/<typename> - Flat list - expanded, Date - oldest first, Display below post or comments.
  - You will need to do this for each content type that has comments enabled.
- (performance) remove line 76 in screen.css : .showgrid {background:url(src/grid.png);}  
  this saves an uncessary HTTP request on your server
- (performance) apply system.css.patch to remove uncessary HTTP requests to images that SimplyCivi overrides already



--- USING SimplyCivi --------------------------------------------------------

SimplyCivi aligns designs to a grid. If you add a class "showgrid" to any <div class="container showgrid"> it will
show you the grid it is working with, both in terms of columns and rows. You can read more by following the links
on the homepage: http://www.SimplyCivicss.org/

To get SimplyCivi to work with Drupal (because Drupal adds in paddings, borders, and more), we need to override
some values in Drupal and recalculate others so that it more closely follows the grid. Here are how such
calculations work:

# Default values
  - Browsers default font-size: 16px
  - Base font-size: 75% = 12px = 1em
  - Base line-height: 1.5em = 18px

# Calculating font size
  * font-size (em) to font-size (px):
    base * relative = result
    12px * 0.9em    = 10.8px

  * font-size (px) to font-size (em):
    font-size / base font-size = relative font-size
    15px      / 12px           = 1.25em

# Calculating line height
  * font-size (px) to line-height (em):
    base line-height / font-size = line-height
    18px             / 10px      = 1.8em

    * font-size (em) to line-height (em):
    base line-height / (base font-size * font-size) = line-height
    18px             / (12px           * 0.9em    ) = 1.667em

# Calculating pixel size
  1 / font-size = 1 pixel in ems
  1 / 12px      = 0.0834em

# Using horizontal borders
  The box model states that border width is added to the dimensions of the box. This means whenever a top or bottom border
  is used, we must account for the extra height by decreasing the padding.

  Example

  Here we need to adjust the padding-bottom value to account for the extra pixel added by the border-bottom:

    .class {
      border-bottom: 1px solid #ccc;
      padding-bottom: 1.5em;   /* WRONG!! */
    }

  The line height is 18px and putting a padding-bottom of 1.5em gives us a nice full line break below. But because we
  have a border too, we need to do slightly less than 1.5em (e.g., 18px - 1px):

    required pixels / base-font = em value
    17px            / 12px      = 1.4167em

    .class {
      border-bottom: 1px solid #ccc;
      padding-bottom: 1.4167em;   /* CORRECT :) */
    }


--- NOTE: -----------------------------------------------------------
SimplyCivi theme now includes two ways of adding a logo: via the $logo variable or via css. There is a setting in the theme settings to choose your method.

If you use the css method, here are some tips for adding it.

h1 a{
width: [width of logo];
height: [height of logo];
display:block;
text-indent: -99999em;
background: url(../images/logo.png) no-repeat left top;
}

What this ends up doing is removing the text title of the site, and replacing it with a background image of the logo that is still linked to the homepage of the site.



CHANGELOG.txt
--- CHANGELOG --------------------------------------------------------
2.x, 2010-06-20
----------------------
- Fixed: Adjusted function that displays breadcrumbs to support menu_breadcrumb. [#789210]
- Fixed: Added html_entity_decode() to the SimplyCivi_trim_text() functions to ensure that the entities don't show up in the headers. [#806608] 

2.x, 2010-05-20
----------------------
- Fixed: Bug in the code with an elseif instead of an if at line 54.

2.x, 2010-05-18
----------------------
- Removed: Removed the code that added "" marks around the taxonomy pages. This can be easily over written via the Views module.
- Fixed: Enable breadcrumbs on items with one or more links.

2.x, 2010-04-19
----------------------
- Fixed: Some title tags included html. That is now being removed. [#769778] [Thanks madt & amorfati]

2.x, 2010-03-22
----------------------
- Fixed: Comment body width was not correct, especially in Firefox. [#466360]
- Fixed: Removed class for 'no-picture' if there is no user picture for comments. It is just redundant as you can set the default picture at /admin/user/settings. 


2.x, 2010-03-13
----------------------
- Fixed: Cleaned up usage of format_plural() in SimplyCivi_comment_wrapper(). [#741034] [Thanks madt.]

2.x, 2010-03-03
----------------------
- Added: .make file to download the css framework from GitHub.

2.x, 2010-03-03
----------------------
- Fixed: RTL style were not in place. Now they are!

2.x, 2010-02-26
----------------------
- Fixed: A stray DPM that got left behind.

2.x, 2010-02-24
----------------------
- Fixed: Fix path to the IE stylesheets in subthemes. Added a new variable $path_parent that returns the path to SimplyCivi.

2.x, 2010-02-24
----------------------
- Fixed: Lingering issue with default variables not getting set correctly in the template.theme-registry.inc causing the global defaults to not stick. [#705058][Thanks zdean for alerting me to this issue.]

2.x, 2010-02-16
----------------------
- Added: better class names for the primary and secondary navigation bars.

2.x, 2010-02-16
----------------------
- Fixed: README.txt had incorrect path listed in the installation help. (Thanks fm for pointing this out.)

2.x, 2010-02-10
----------------------
- Fixed: When "Site Name" gets turned off via the theme settings, the titles get built wrong.

2.x, 2010-02-08
----------------------
- Fixed: Re-factored the code the displays the header (logo, site name, etc) to fully use the default phptemplate settings. [#702964] [Thanks GreenSkin and zdean for the help on this.]

2.x, 2010-02-06
----------------------
- Fixed: Issue where turning off the site name via the drupal built in settings causes the site logo to be non-linkable. [#702964]

2.x, 2010-02-02
----------------------
- Added: Moved changelog to CHANGELOG.txt
- Moved: Move the SimplyCivi CSS files from the '/SimplyCivi/' folder to '/framework/' [#660224]
- Added: SimplyCivi_css_reorder() function to ensure that the SimplyCivi files stay at the top of the css stack
- Fixed: Added a temporary fix to the SimplyCiviCSS that makes the input boxes not aligned properly. [#700464]
- Fixed: A bug that was putting the BPSTARTER theme css too low in the stack. [#702852]

2.x, 2010-01-21
----------------------
- Fixed: Links now show up on comments for non-admin users. [#682676]
- Added: Adding support for RTL sites. This is a work in progress but commiting it to keep it moving [#666220]
- Fixed: Comment subjects now reacting properly to comment_subject_field_ variables.

2.x, 2010-01-08
----------------------
- Fixed: Replaced hard tabs with 2 spaces.
- Fixed: #665428 by Scott Reynolds cleanup of php notice messages and path calls.

2.x, 2010-01-04
----------------------
- Add column variables to the BPSTARTER theme for redefining region sizes.

2.x, 2009-12-30
----------------------
- Add option to disable fancy javascript that removes the submit button after you have pressed it to prevent duplicate posts. Was conflicting with the autocomplete.js. [#579070]
- Renaming Sidebar labels to "First Sidebar" and "Second Sidebar".

2.x, 2009-12-28
----------------------
- Fixing incompatibilities with Page Title module. [#661422]

2.x , 2009-12-14
----------------------
-  A variety of bug fixes submitted by Scott Reynolds: [#658270] [#658818] and  [#658322]

2.x, 2009-12-08
----------------------
- Reorganized files
- Added Starter sub-theme
- Added hook_theme function and functions to provide theme settings as well as the ability to pass those down to sub-themes.


1.x, 2008-12-08
----------------------
- Fix sidebar variables so they don't build unnecessary nested divs when using sub-themes.

1.x, 2008-10-01
----------------------
- add admin links in each block on hover to edit and configure the blocks as well as edit the menu blocks.

1.x, xxxx-xx-xx
----------------------
- add permalinks to each comment

1.2, 2008-09-04
----------------------
- add missing header region
- add missing primary and secondary links

1.1, 2008-08-26
----------------------
- Don't allow search engines to index duplicate pages created by the pager
  More: http://www.seo-expert-blog.com/blog/avoiding-duplicate-title-tags-on-pager-pages-in-drupal
- Show node body as meta description if teaser not available


1.0, 2008-06-04
----------------------

- Initial 1.0 release


template.php
<?php
// $Id: template.php,v 1.15.2.1.2.20.2.1 2009/12/07 19:13:20 designerbrent Exp $

// Auto-rebuild the theme registry during theme development.
if (theme_get_setting('SimplyCivi_rebuild_registry')) {
  drupal_rebuild_theme_registry();
}
if (theme_get_setting('SimplyCivi_animated_submit')) {
  drupal_add_js(drupal_get_path('theme', 'SimplyCivi') .'/scripts/submit_animated.js');
}

/**
 * Implements HOOK_theme().
 */
function SimplyCivi_theme(&$existing, $type, $theme, $path) {
  if (!db_is_active()) {
    return array();
  }
  include_once './' . drupal_get_path('theme', 'SimplyCivi') . '/template.theme-registry.inc';
  return _SimplyCivi_theme($existing, $type, $theme, $path);
}

/**
 * Intercept page template variables
 *
 * @param $vars
 *   A sequential array of variables passed to the theme function.
 */
function SimplyCivi_preprocess_page(&$vars) {
  global $user;
  $vars['path'] = base_path() . path_to_theme() .'/';
  $vars['path_parent'] = base_path() . drupal_get_path('theme', 'SimplyCivi') . '/';
  $vars['user'] = $user;

  // Prep the logo for being displayed
  $site_slogan = (!$vars['site_slogan']) ? '' : ' - '. $vars['site_slogan'];
  $logo_img ='';
  $title = $text = variable_get('site_name', '');
  if ($vars['logo']) {
    $logo_img = "<img src='". $vars['logo'] ."' alt='". $title ."' border='0' />";
    $text = ($vars['site_name']) ? $logo_img . $text : $logo_img;
  }
  $vars['logo_block'] = (!$vars['site_name'] && !$vars['logo']) ? '' : l($text, '', array('attributes' => array('title' => $title . $site_slogan), 'html' => !empty($logo_img)));
  // Even though the site_name is turned off, let's enable it again so it can be used later.
  $vars['site_name'] = variable_get('site_name', '');

  //Play nicely with the page_title module if it is there.
  if (!module_exists('page_title')) {
    // Fixup the $head_title and $title vars to display better.
    $title = drupal_get_title();
    $headers = drupal_set_header();

    // if this is a 403 and they aren't logged in, tell them they need to log in
    if (strpos($headers, 'HTTP/1.1 403 Forbidden') && !$user->uid) {
      $title = t('Please login to continue');
    }
    $vars['title'] = $title;

    if (!drupal_is_front_page()) {
      $vars['head_title'] = $title .' | '. $vars['site_name'];
      if ($vars['site_slogan'] != '') {
        $vars['head_title'] .= ' &ndash; '. $vars['site_slogan'];
      }
    }
    $vars['head_title'] = strip_tags($vars['head_title']);
  }

  //Perform RTL - LTR swap and load RTL Styles.
  if ($vars['language']->dir == 'rtl') {
    // Remove SimplyCivi Grid and use RTL grid
    $css = $vars['css'];
    $css['screen,projection']['theme'][path_to_theme() .'/SimplyCivi/SimplyCivi/plugins/rtl/screen.css'] = TRUE;
    $vars['styles'] = drupal_get_css($css);
    
    //setup rtl css for IE
    $vars['styles_ie']['ie'] = '<link href="'. $path .'css/ie-rtl.css" rel="stylesheet"  type="text/css"  media="screen, projection" />';
    $vars['styles_ie']['ie6'] = '<link href="'. $path .'css/ie6-rtl.css" rel="stylesheet"  type="text/css"  media="screen, projection" />';
  }

  $vars['meta'] = '';
  // SEO optimization, add in the node's teaser, or if on the homepage, the mission statement
  // as a description of the page that appears in search engines
  if ($vars['is_front'] && $vars['mission'] != '') {
    $vars['meta'] .= '<meta name="description" content="'. SimplyCivi_trim_text($vars['mission']) .'" />'."\n";
  }
  elseif (isset($vars['node']->teaser) && $vars['node']->teaser != '') {
    $vars['meta'] .= '<meta name="description" content="'. SimplyCivi_trim_text($vars['node']->teaser) .'" />'."\n";
  }
  elseif (isset($vars['node']->body) && $vars['node']->body != '') {
    $vars['meta'] .= '<meta name="description" content="'. SimplyCivi_trim_text($vars['node']->body) .'" />'."\n";
  }
  // SEO optimization, if the node has tags, use these as keywords for the page
  if (isset($vars['node']->taxonomy)) {
    $keywords = array();
    foreach ($vars['node']->taxonomy as $term) {
      $keywords[] = $term->name;
    }
    $vars['meta'] .= '<meta name="keywords" content="'. implode(',', $keywords) .'" />'."\n";
  }

  // SEO optimization, avoid duplicate titles in search indexes for pager pages
  if (isset($_GET['page']) || isset($_GET['sort'])) {
    $vars['meta'] .= '<meta name="robots" content="noindex,follow" />'. "\n";
  }

  if (theme_get_setting('SimplyCivi_showgrid')) {
    $vars['body_classes'] .= ' showgrid ';
  }

  // Make sure framework styles are placed above all others.
  $vars['css_alt'] = SimplyCivi_css_reorder($vars['css']);
  $vars['styles'] = drupal_get_css($vars['css_alt']);

  /* I like to embed the Google search in various places, uncomment to make use of this
  // setup search for custom placement
  $search = module_invoke('google_cse', 'block', 'view', '0');
  $vars['search'] = $search['content'];
  */

  /* to remove specific CSS files from modules use this trick
  // Remove stylesheets
  $css = $vars['css'];
  unset($css['all']['module']['sites/all/modules/contrib/plus1/plus1.css']);
  $vars['styles'] = drupal_get_css($css);
  */

}

/**
 * Intercept node template variables
 *
 * @param $vars
 *   A sequential array of variables passed to the theme function.
 */
function SimplyCivi_preprocess_node(&$vars) {
  $node = $vars['node']; // for easy reference
  // for easy variable adding for different node types
  switch ($node->type) {
    case 'page':
      break;
  }
}

/**
 * Intercept comment template variables
 *
 * @param $vars
 *   A sequential array of variables passed to the theme function.
 */
function SimplyCivi_preprocess_comment(&$vars) {
  static $comment_count = 1; // keep track the # of comments rendered
  // Calculate the comment number for each comment with accounting for pages.
  $page = 0;
  $comments_previous = 0;
  if (isset($_GET['page'])) {
    $page = $_GET['page'];    
    $comments_per_page = variable_get('comment_default_per_page_' . $vars['node']->type, 1);
    $comments_previous = $comments_per_page * $page;
  }
  $vars['comment_count'] =  $comments_previous + $comment_count;

  // if the author of the node comments as well, highlight that comment
  $node = node_load($vars['comment']->nid);
  if ($vars['comment']->uid == $node->uid) {
    $vars['author_comment'] = TRUE;
  }

  // If comment subjects are disabled, don't display them.
  if (variable_get('comment_subject_field_' . $vars['node']->type, 1) == 0) {
    $vars['title'] = '';
  }
  else {
    $vars['title'] = l($vars['comment']->subject, 'node/'. $vars['node']->nid, array('query' => $query, 'fragment' => $fragment));
  }

  // Add the pager variable to the title link if it needs it.
  $fragment = 'comment-' . $vars['comment']->cid;
  $query = '';
  if (!empty($page)) {
    $query = 'page='. $page;
  }

  $vars['comment_count_link'] = l('#'. $vars['comment_count'], 'node/'. $vars['node']->nid, array('query' => $query, 'fragment' => $fragment));

  $comment_count++;
}

/**
 * Override or insert variables into the block templates.
 *
 * @param $vars
 *   An array of variables to pass to the theme template.
 * @param $hook
 *   The name of the template being rendered ("block" in this case.)
 */
function SimplyCivi_preprocess_block(&$vars, $hook) {
  $block = $vars['block'];

  // Special classes for blocks.
  $classes = array('block');
  $classes[] = 'block-' . $block->module;
  $classes[] = 'region-' . $vars['block_zebra'];
  $classes[] = $vars['zebra'];
  $classes[] = 'region-count-' . $vars['block_id'];
  $classes[] = 'count-' . $vars['id'];

  $vars['edit_links_array'] = array();
  $vars['edit_links'] = '';

  if (theme_get_setting('SimplyCivi_block_edit_links') && user_access('administer blocks')) {
    include_once './' . drupal_get_path('theme', 'SimplyCivi') . '/template.block-editing.inc';
    SimplyCivi_preprocess_block_editing($vars, $hook);
    $classes[] = 'with-block-editing';
  }

  // Render block classes.
  $vars['classes'] = implode(' ', $classes);
}


/**
 * Intercept box template variables
 *
 * @param $vars
 *   A sequential array of variables passed to the theme function.
 */
function SimplyCivi_preprocess_box(&$vars) {
  // rename to more common text
  if (strpos($vars['title'], 'Post new comment') === 0) {
    $vars['title'] = 'Add your comment';
  }
}

/**
 * Override, remove "not verified", confusing
 *
 * Format a username.
 *
 * @param $object
 *   The user object to format, usually returned from user_load().
 * @return
 *   A string containing an HTML link to the user's page if the passed object
 *   suggests that this is a site user. Otherwise, only the username is returned.
 */
function SimplyCivi_username($object) {
  if ($object->uid && $object->name) {
    // Shorten the name when it is too long or it will break many tables.
    if (drupal_strlen($object->name) > 20) {
      $name = drupal_substr($object->name, 0, 15) .'...';
    }
    else {
      $name = $object->name;
    }

    if (user_access('access user profiles')) {
      $output = l($name, 'user/'. $object->uid, array('attributes' => array('title' => t('View user profile.'))));
    }
    else {
      $output = check_plain($name);
    }
  }
  elseif ($object->name) {
    // Sometimes modules display content composed by people who are
    // not registered members of the site (e.g. mailing list or news
    // aggregator modules). This clause enables modules to display
    // the true author of the content.
    if (!empty($object->homepage)) {
      $output = l($object->name, $object->homepage, array('attributes' => array('rel' => 'nofollow')));
    }
    else {
      $output = check_plain($object->name);
    }
  }
  else {
    $output = variable_get('anonymous', t('Anonymous'));
  }

  return $output;
}

/**
 * Override, make sure Drupal doesn't return empty <P>
 *
 * Return a themed help message.
 *
 * @return a string containing the helptext for the current page.
 */
function SimplyCivi_help() {
  $help = menu_get_active_help();
  // Drupal sometimes returns empty <p></p> so strip tags to check if empty
  if (strlen(strip_tags($help)) > 1) {
    return '<div class="help">'. $help .'</div>';
  }
}

/**
 * Override, use a better default breadcrumb separator.
 *
 * Return a themed breadcrumb trail.
 *
 * @param $breadcrumb
 *   An array containing the breadcrumb links.
 * @return a string containing the breadcrumb output.
 */
function SimplyCivi_breadcrumb($breadcrumb) {
  // Don't add the title if menu_breadcrumb exists. TODO: Add a settings 
  // checkbox to optionally control the display.
  if (!module_exists(menu_breadcrumb) && count($breadcrumb) > 0) {
      $breadcrumb[] = drupal_get_title();
  }
  return '<div class="breadcrumb">'. implode(' &rsaquo; ', $breadcrumb) .'</div>';
}

/**
 * Rewrite of theme_form_element() to suppress ":" if the title ends with a punctuation mark.
 */
function SimplyCivi_form_element($element, $value) {
  $args = func_get_args();
  return preg_replace('@([.!?]):\s*(</label>)@i', '$1$2', call_user_func_array('theme_form_element', $args));
}

/**
 * Set status messages to use SimplyCivi CSS classes.
 */
function SimplyCivi_status_messages($display = NULL) {
  $output = '';
  foreach (drupal_get_messages($display) as $type => $messages) {
    // SimplyCivi can either call this success or notice
    if ($type == 'status') {
      $type = 'success';
    }
    $output .= "<div class=\"messages $type\">\n";
    if (count($messages) > 1) {
      $output .= " <ul>\n";
      foreach ($messages as $message) {
        $output .= '  <li>'. $message ."</li>\n";
      }
      $output .= " </ul>\n";
    }
    else {
      $output .= $messages[0];
    }
    $output .= "</div>\n";
  }
  return $output;
}

/**
 * Override comment wrapper to show you must login to comment.
 */
function SimplyCivi_comment_wrapper($content, $node) {
  global $user;
  $output = '';

  if ($node = menu_get_object()) {
    if ($node->type != 'forum') {
      $count = ($node->comment_count > 0) ? format_plural($node->comment_count, '1 comment', '@count comments') : t('No comments available.');
      $output .= '<h3 id="comment-number">'. $count .'</h3>';
    }
  }

  $output .= '<div id="comments">';
  $msg = '';
  if (!user_access('post comments')) {
    $dest = 'destination='. $_GET['q'] .'#comment-form';
    $msg = '<div id="messages"><div class="error-wrapper"><div class="messages error">'. t('Please <a href="!register">register</a> or <a href="!login">login</a> to post a comment.', array('!register' => url("user/register", array('query' => $dest)), '!login' => url('user', array('query' => $dest)))) .'</div></div></div>';
  }
  $output .= $content;
  $output .= $msg;

  return $output .'</div>';
}

/**
 * Override, use better icons, source: http://drupal.org/node/102743#comment-664157
 *
 * Format the icon for each individual topic.
 *
 * @ingroup themeable
 */
function SimplyCivi_forum_icon($new_posts, $num_posts = 0, $comment_mode = 0, $sticky = 0) {
  // because we are using a theme() instead of copying the forum-icon.tpl.php into the theme
  // we need to add in the logic that is in preprocess_forum_icon() since this isn't available
  if ($num_posts > variable_get('forum_hot_topic', 15)) {
    $icon = $new_posts ? 'hot-new' : 'hot';
  }
  else {
    $icon = $new_posts ? 'new' : 'default';
  }

  if ($comment_mode == COMMENT_NODE_READ_ONLY || $comment_mode == COMMENT_NODE_DISABLED) {
    $icon = 'closed';
  }

  if ($sticky == 1) {
    $icon = 'sticky';
  }

  $output = theme('image', path_to_theme() . "/images/icons/forum-$icon.png");

  if ($new_posts) {
    $output = "<a name=\"new\">$output</a>";
  }

  return $output;
}

/**
 * Override, remove previous/next links for forum topics
 *
 * Makes forums look better and is great for performance
 * More: http://www.sysarchitects.com/node/70
 */
function SimplyCivi_forum_topic_navigation($node) {
  return '';
}

/**
 * Trim a post to a certain number of characters, removing all HTML.
 */
function SimplyCivi_trim_text($text, $length = 150) {
  // remove any HTML or line breaks so these don't appear in the text
  $text = trim(str_replace(array("\n", "\r", "\r\n"), ' ', strip_tags(html_entity_decode($text, ENT_QUOTES, 'UTF-8'))));
  $text = trim(substr($text, 0, $length));
  $lastchar = substr($text, -1, 1);
  // check to see if the last character in the title is a non-alphanumeric character, except for ? or !
  // if it is strip it off so you don't get strange looking titles
  if (preg_match('/[^0-9A-Za-z\!\?]/', $lastchar)) {
    $text = substr($text, 0, -1);
  }
  // ? and ! are ok to end a title with since they make sense
  if ($lastchar != '!' && $lastchar != '?') {
    $text .= '...';
  }
  return $text;
}

/**
 * This rearranges how the style sheets are included so the framework styles
 * are included first.
 *
 * Sub-themes can override the framework styles when it contains css files with
 * the same name as a framework style. This can be removed once Drupal supports
 * weighted styles.
 */
function SimplyCivi_css_reorder($css) {
  global $theme_info, $base_theme_info;

  // Dig into the framework .info data.
  $framework = !empty($base_theme_info) ? $base_theme_info[0]->info : $theme_info->info;

  // Pull framework styles from the themes .info file and place them above all stylesheets.
  if (isset($framework['stylesheets'])) {
    foreach ($framework['stylesheets'] as $media => $styles_from_bp) {
      // Setup framework group.
      if (isset($css[$media])) {
        $css[$media] = array_merge(array('framework' => array()), $css[$media]);
      }
      else {
        $css[$media]['framework'] = array();
      }
      foreach ($styles_from_bp as $style_from_bp) {
        // Force framework styles to come first.
        if (strpos($style_from_bp, 'framework') !== FALSE) {
          $framework_shift = $style_from_bp;
          $remove_styles = array($style_from_bp);
          // Handle styles that may be overridden from sub-themes.
          foreach ($css[$media]['theme'] as $style_from_var => $preprocess) {
            if ($style_from_bp != $style_from_var && basename($style_from_bp) == basename($style_from_var)) {
              $framework_shift = $style_from_var;
              $remove_styles[] = $style_from_var;
              break;
            }
          }
          $css[$media]['framework'][$framework_shift] = TRUE;
          foreach ($remove_styles as $remove_style) {
            unset($css[$media]['theme'][$remove_style]);
          }
        }
      }
    }
  }

  return $css;
}

